scripts/lang_adapt/madx_run_clm.py

```diff
@@ -711,11 +834,50 @@ def main():
     if training_args.do_eval:
         eval_dataset = lm_datasets["validation"]
     
+    # compute K value for SFT (https://arxiv.org/pdf/2110.07560.pdf)
+    if adapter_args.train_sft and not adapter_args.train_adapter:
+        # override the K value if adapter_reduction_factor is set
+        if adapter_args.adapter_reduction_factor:
+            logger.info(f"Overriding K value for SFT with adapter_reduction_factor: {adapter_args.adapter_reduction_factor}")
+            # calc appropriate K value
+            num_layers = len(model.transformer.h)
+            sft_k = num_layers * model.transformer.word_embeddings.weight.shape[1] ** 2 // adapter_args.adapter_reduction_factor * 2 #* 2 for the up and down proj
+
+            sft_k += model.transformer.word_embeddings.weight.shape[1]  ** 2 // 2 # inv adapters. TODO: if we use other adapter configs, this breaks (code works, but K no longer matches adapter budget)
+
+            adapter_args.ft_params_num = int(sft_k)
+            logger.info(f"K value for SFT is {adapter_args.ft_params_num}")
```

Comment:
Can I move this line 838 - 849 into the condition after line 702?

Reply:
Yes! Also, does the K value computed here match what the # of adapter parameters should be?